op-devstack/sysgo/l2_challenger.go

```diff
@@ -39,13 +39,17 @@ func WithL2Challenger(challengerID stack.L2ChallengerID, l1ELID stack.L1ELNodeID
 	supervisorID *stack.SupervisorID, clusterID *stack.ClusterID, l2CLID *stack.L2CLNodeID, l2ELIDs []stack.L2ELNodeID,
 ) stack.Option[*Orchestrator] {
 	return stack.AfterDeploy(func(orch *Orchestrator) {
-		require := orch.P().Require()
+		ctx := orch.P().Ctx()
+		ctx = stack.ContextWithKind(ctx, stack.L2ChallengerKind)
+		p := orch.P().WithCtx(ctx)
+
+		require := p.Require()
 		require.False(orch.challengers.Has(challengerID), "challenger must not already exist")
 
 		challengerSecret, err := orch.keys.Secret(devkeys.ChallengerRole.Key(l1ELID.ChainID.ToBig()))
 		require.NoError(err)
 
-		logger := orch.P().Logger().New("service", "op-challenger", "id", challengerID)
+		logger := p.Logger().New("service", "op-challenger", "id", challengerID)
```

Comment:
what goes into the decision to attach these attrs to the logger directly rather than via the context?

Reply:
Oh I missed this one, thanks for catching it.